cranelift/wasmtests/if-unreachable-else-params-2.wat

```diff
@@ -0,0 +1,75 @@
+(module
+  (type (;0;) (func (param i32)))
+  (func $main (type 0) (param i32)
+    i32.const 39
+    local.get 0
+    if (param i32)  ;; label = @1
+      if  ;; label = @2
+      end
+      i32.const 39
+      local.get 0
+      if (param i32)  ;; label = @2
+        if  ;; label = @3
+        end
+        i32.const 0
+        i32.popcnt
+        i32.popcnt
+        i32.popcnt
+        if  ;; label = @3
+          unreachable
+        end
+      else
+        i32.load16_s align=1
+        i64.load32_s align=1
+        unreachable
+        unreachable
+        unreachable
+        unreachable
+        f64.lt
+        unreachable
+        unreachable
+        unreachable
+        unreachable
+        unreachable
```

Comment:
Are all those duplicate `unreachable` necessary to reproduce the bug?

Reply:
Actually, I can't seem to get it to reproduce with `clif-util` at all, now that I play with this -- the panic only manifests in the SpiderMonkey embedding (likely due to some subtle difference in the FuncEnvironment, perhaps? The error that occurs is an I32/I64 mismatch so it's not guaranteed to occur...). I've removed the test case, but I can work out a better way to test this if needed.

Reply:
I've updated the PR to include a repro case that I constructed starting from another (`if-unreachable-else-params.wat`), so this is tested now independently of SpiderMonkey.